doc/rfcs/2023-08-17--ManagingVersions.md

```diff
@@ -0,0 +1,199 @@
+        tiny busy-work style things we need to 
+ - It's off-putting to new users
+    - Some of the latent consistency issues are (rightly) tagged a
+        `good-first-issue`, yet when folks make PRs to fix them
+        (#3397, #3613, etc) we reject the changes.
+ - We throw away good code and developer context
+    - This follows from the last two points---we should fix things when
+        we are touching the code to make changes. Otherwise when we come
+        back later during release crunch time, we're relearning context
+        of what needs fixing and how to fix it that was already in
+        someone's head a few months ago.
+
+
+### What can we do instead?
+
+1. Keep PRs with fixes tagged and open until a major release process starts.
+   - Pro: Have the "real code" and "real fix" already there.
+   - Con: Likely that the PR will not merge cleanly by the time the next
+       major release occurs.
+   - Con: Lots of noise in the PR queue
+
+2. Always keep a 'next-major-release' branch active
+   - Pro: Logical home for ABI-breaking changes
+   - Pro: Developer-friendly, easy to ask people to change target of PR
+       to accept useful changes
+   - Con: Will diverge, likely significantly from master branch, making
+       sync hard when release is ready
+   - Con: Avoiding prior con would add non-trivial maintenance burden of
+       periodically merging master with 'next-major', and likely
+       carrying a large set of merge conflict resolutions along the way.
+   - Con: Somewhat hidden, and unlikely to see any testing
+
+3. The `cfg` option (I can't believe I'm suggesting it either)
+   - Pro: Code is 'right there' such that wide-area interface,
+       renaming, etc style changes will update next-release code as well
+       [assumes CI builds both current and next-release, which is easy]
+   - Pro: Release transition is very clear, simply remove all code where
+       `cfg` is no longer relevant.
+   - Con: It's `cfg`. It means we're carrying a bunch of code which in
+       the common case is untested and unused.
+   - Pro: Can result in space-savings for up-to-date downstream users.
+       If we are carrying code for backwards compatibility that
+       downstream folks know they won't use, they can remove that part
+       of the kernel.
+       - Con: The exponential number of kernel versions we just created.
+       - Pro: Could act as indicator for when a release is merited, i.e.
+           if we released a new version we'd shave off XX% of kernel
+           size for large enough X.
+   - Nuance: It may be worth distinguishing between "next minor" and
+       "next major". In particular, "next minor" `cfg`'d stuff can be
+       default-on, and largely no change from current practice, except
+       for this sentinel sprinkled in the codebase identifying what's
+       not in the released version [though, this could arguably also be
+       accomplished with some type of comment keyword].
+
+4. ...?
```

Comment:
```suggestion
4. Open a tracking issue for the change needed for the next release. Include the diff required to implement the change in the issue text.
    - Pro: Work to implement the change already done.
    - Pro: Because by definition this is a change to some stabilized portion of the kernel, the patch is likely to cleanly apply or be very close when the next release is prepared.
    - Con: Someone has to remember to address the tracking issue.
```